Replace deprecated StringPtr function

kei01234kei opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
I replace deprecated `StringPtr` into `String`.
ref: https://github.com/kubernetes/kubernetes/pull/114848
#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

Jefftree commented:
See https://github.com/kubernetes/kubernetes/blob/master/vendor/k8s.io/utils/pointer/pointer.go#L16-L19. Package is deprecated, should use the ptr library instead.

/ok-to-test

aaron-prindle commented:
/triage accepted

kei01234kei committed:
pointer.String to ptr.To

kei01234kei commented:
> See https://github.com/kubernetes/kubernetes/blob/master/vendor/k8s.io/utils/pointer/pointer.go#L16-L19. Package is deprecated, should use the ptr library instead.

Thanks. Modified
[pointer.String to ptr.To](https://github.com/kubernetes/kubernetes/pull/130722/commits/b6546211284e99af832057c32eb38d73e5553217)

kei01234kei commented:
/retest

Jefftree commented:
/lgtm
/assign @deads2k @richabanker

richabanker commented:
/approve 
for `staging/src/k8s.io/component-base/metrics`  changes

kei01234kei commented:
@deads2k 
Please review this pr if u have time 🙏

kei01234kei committed:
Replace deprecated StringPtr function

# Conflicts:
#	staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

kei01234kei committed:
chore

kei01234kei commented:
/close
